pyext/src/macros.py

```diff
@@ -568,14 +600,33 @@ def execute_macro(self):
                         output.write_stat2(low_temp_stat_file)
                 ntimes_at_low_temp += 1
 
-            if not self.test_mode:
+            if not self.test_mode and not self.nest:
                 output.write_stat2(replica_stat_file)
             if self.vars["replica_exchange_swap"]:
                 rex.swap_temp(i, score)
+
+        if self.nest and len(sampled_likelihoods) > 0:
+            with open(
+                "likelihoods_"+ \
+                str(self.replica_exchange_object.get_my_index()),"wb") as lif:
+                pickle.dump(sampled_likelihoods, lif)
+
+            nestor_rmf_fname = str(self.nestor_rmf_fname)+ '_'+\
+                str(self.replica_exchange_object.get_my_index())+'.rmf3'
+
+            if nestor_rmf_fname not in os.listdir(os.getcwd()):
```

Comment:
This seems like a very odd way of saying `if not os.path.exists(nestor_rmf_fname):` (and inefficient, since it must potentially stat every file in the working directory).

Reply:
This part got updated and now it does not do this check